packages/common/client-utils/src/bufferNode.ts

```diff
@@ -46,9 +46,13 @@ export type IsoBuffer = Buffer;
  *
  * @internal
  */
-export function Uint8ArrayToString(arr: Uint8Array, encoding?: string): string {
-	// Make this check because Buffer.from(arr) will always do a buffer copy
-	return Buffer.isBuffer(arr) ? arr.toString(encoding) : Buffer.from(arr).toString(encoding);
+export function Uint8ArrayToString(
+	arr: Uint8Array,
+	encoding?: "utf8" | "utf-8" | "base64",
+): string {
+	// Buffer extends Uint8Array.  Therefore, 'arr' may already be a Buffer, in
+	// which case we can avoid copying the Uint8Array into a new Buffer instance.
+	return (Buffer.isBuffer(arr) ? arr : Buffer.from(arr)).toString(encoding);
```

Comment:
When written this way, TypeScript misses that .isBuffer(..) implies arr is an instance of buffer:
```ts
Buffer.isBuffer(arr)
    ? arr.toString(...)                         // <== Error: type of 'arr' is 'Uint8Array'.
     : Buffer.from(arr).toString(...)
```
The fix is to rearrange the expression as so:
```ts
    (Buffer.isBuffer(arr)
        ? arr
        : Buffer.from(arr))    // <== Expresion has type 'Buffer'
    .toString(..)
```